### PR TITLE
Introduce new configurable information panel to the placeholder

### PR DESCRIPTION
**Pull request by i-like-robots**
Capable of showing video title, description brand and video duration.

screen shot 2016-09-12 at 11 03 26

Drops the previous `placeholdertitle` option and replaces it with a new `placeholderInfo` option that accepts an array of features (brand, duration, title and description). I guess this means a new major version.

This change _also_ exposes some utility classes and mixins that can alter the presentation of the info panel depending on its size in context:

screen shot 2016-09-12 at 15 51 07

So an implementation may add the `o-video--{size}` class name to the element _or_ use the mixins if the video element is not a static size:

``` scss
.o-video {
  @include oVideoInfoSmall;

  @include oGridRespondTo('M') {
    @include oVideoInfoMedium;
  }
}
```

Tested in FF, Chrome and IE9

### CODE REVIEW

File: origami.json [hunk 1]
```diff
@@ -11,13 +11,14 @@
...
 		"js": "demos/src/demo.js",
-		"template": "demos/src/video.mustache"
+		"dependencies": "o-fonts@^2.1.0"
 	},
 	"demos": [
 		{
 			"name": "native",
 			"description": "Native",
-			"expanded": true
+			"expanded": true,
```
Comment (JakeChampion) [inline origami.json]: This property is deprecated and should be removed. The registry no longer listens to this property.

File: origami.json [hunk 2]
```diff
@@ -31,6 +32,13 @@
 			"expanded": true,
 			"template": "demos/src/playlist.mustache",
 			"js": "demos/src/playlist.js"
+		},
+		{
+			"name": "size variations",
+			"description": "Display the video at preset sizes",
+			"expanded": true,
```
Comment (JakeChampion) [inline origami.json]: Same here, this can be removed.

File: src/js/info.js [hunk 3]
```diff
@@ -0,0 +1,86 @@
+function formatBrand (tags) {
+	const regex = /brand:/i;
+	const tag = tags && tags.find(tag => regex.test(tag));
```
Comment (JakeChampion) [inline src/js/info.js:3]: `Array.prototype.find` is a new method and should be added the the `browserFeatures.required` array in `origami.json`.

File: src/js/video.js [hunk 4]
```diff
@@ -66,26 +71,23 @@ const defaultOpts = {
...
 		this.containerEl = el;
 
-		this.opts = opts || getOptionsFromDataAttributes(this.containerEl.attributes);
-
-		Object.keys(defaultOpts).forEach(optionName => {
-			if (typeof this.opts[optionName] === 'undefined') {
-				this.opts[optionName] = defaultOpts[optionName];
-			}
-		});
+		this.opts = Object.assign({}, defaultOpts, opts, getOptionsFromDataAttributes(this.containerEl.attributes));
```
Comment (JakeChampion) [inline src/js/video.js:82]: `Object.assign` should probably be in the `browserFeatures.required` array in `origami.json`.

File: README.md [hunk 5]
```diff
@@ -34,7 +33,7 @@ Where `opts` is an optional object with properties
  * `optimumvideowidth` [`Number`] The optimum width of the video itself, used when there are multiple video renditions available to
  decide which to display (the smallest one that's at least as large as this width, if it exists)
  * `placeholder` [`Boolean`] Show just the poster image, load (and play) video on click
- * `placeholdertitle` [`Boolean`] Show just the title as an overlay on the placeholder
+ * `placeholderInfo` [`Array`] A list of extra information to display on the placeholder (Available: title, description, brand, duration)
```
Comment (JakeChampion) [inline README.md:36]: I don't see any tests for the new features, `description`, `brand`, `duration`. Would you be able to add some?

**Comment by JakeChampion** [comment]
I've reviewed this from a technical perspective, looks good.

I can't review this from a design perspective as I have no access to the designs. I assume design has signed off on this and confirmed what is in this PR is what they wanted.

File: README.md (hunk 5 above)
Comment (i-like-robots) [reply README.md:36]: I certainly can

**Comment by i-like-robots** [comment]
Cheers Jake. I had a catch up with Sue but I'll ask her to take a look before we release it to the wider world.

**Comment by i-like-robots** [comment]
Points addressed! This is ready to go AFAIK =]

**Pull request merged**